libs/components/modals/src/lib/modules/modal/modal.component.ts

```diff
@@ -263,6 +316,11 @@ export class SkyModalComponent implements AfterViewInit, OnDestroy, OnInit {
     }
   }
 
+  public ngAfterContentChecked(): void {
+    this.isHeaderEmpty =
+      !this.headingText && !this.headerContent?.nativeElement.children.length;
```

Comment:
This needs like a mutation observer or other change detection handling. It doesn't seem to work for e.g. the `integration` project in default theme.

Reply:
(Steve fixed this with `e2202960bb2931822fef99504a28f8368e860afa`)